src/goInstallTools.ts

```diff
@@ -344,16 +359,22 @@ function installTools(missing: string[]) {
 					resolve([...sofar, null]);
 					return;
 				}
-				cp.execFile(goRuntimePath, ['get', '-u', '-v', allTools[tool]], { env: envForTools }, (err, stdout, stderr) => {
+				let args = ['get', '-u', '-v'];
+				if (tool.endsWith('-gomod')) {
+					args.push('-d');
+				}
+				args.push(allTools[tool]);
+				cp.execFile(goRuntimePath, args, { env: envForTools }, (err, stdout, stderr) => {
 					if (stderr.indexOf('unexpected directory layout:') > -1) {
 						outputChannel.appendLine(`Installing ${tool} failed with error "unexpected directory layout". Retrying...`);
-						cp.execFile(goRuntimePath, ['get', '-u', '-v', allTools[tool]], { env: envForTools }, callback);
+						cp.execFile(goRuntimePath, args, { env: envForTools }, callback);
+					} else if (tool.endsWith('-gomod')) {
```

Comment:
I'll never pass this in my tests. see #2024

Reply:
@Dexus Did you try the latest from master?

You need to update either of `gocode-gomod` or `godef-gomod` via the `Go: Install/Update Tools` command to hit this `else` block

Reply:
As in #2024 already pointed out I fix it myself with a copy. Would be nice if the current fix will be published asap. That works for me.